File: garamana.cpp
#include <iostream>
#include <map>

using namespace std;

int main(){

    map<char, int> conta_letra;

    string palavra_1;
    string palavra_2;

    cin >> palavra_1;
    cin >> palavra_2;
    int i = 0;

    while(palavra_1[i] != '\0'){

        conta_letra[palavra_1[i]]++;

        if (palavra_2[i] != '*'){
            conta_letra[palavra_2[i]]--;
        }
        i++;
    }

    char resp = 'S';
    for(const auto &[letra, qnt]: conta_letra){
        if(qnt < 0){
            resp = 'N';
            break;
        }
    }

    cout << resp << endl;

    return 0;
}

File: codigos/Python/garamana.py
conta_letra = {}

palavra_1 = input()
palavra_2 = input()

i = 0

for i in range(len(palavra_1)):

    letra_1 = palavra_1[i]
    letra_2 = palavra_2[i]

    conta_letra[letra_1] = conta_letra.get(letra_1, 0) + 1

    if letra_2 != '*':
        conta_letra[letra_2] = conta_letra.get(letra_2, 0) - 1

resp = 'S'

for letra, qnt in conta_letra.items():
    if qnt < 0:
        resp = 'N'
        break

print(resp)
